src/repository/opamGit.ml

```diff
@@ -216,7 +214,12 @@ module VCS : OpamVCS.VCS = struct
                    @ List.map OpamFilename.SubPath.to_string
                      (OpamStd.Option.to_list subpath))
     @@> function
-    | { OpamProcess.r_code = 0; _ } -> Done true
+    | { OpamProcess.r_code = 0; _ } ->
+      git repo_root ["submodule"; "status"; "--recursive"] @@> fun r ->
+      if r.r_code = 0 &&
+         List.for_all (fun s -> String.length s > 0 && s.[0] = ' ') r.r_stdout
```

Comment:
It worth a comment here, to explain the whitespace check that is done. The manpage states
```quote
Each SHA-1 will possibly be prefixed with - if the submodule is not initialized,
+ if the currently checked out submodule commit does not match the SHA-1 found in
the index of the containing repository and U if the submodule has merge conflicts.
```
which explains why we require only first character to be a whitespace.